### ISSUE TITLE
Mumble loses link to plugin every few seconds/minutes

### ISSUE THREAD

**Issue by alifeee** [alifeee/Factorio-Proximity-Voice-Chat#9]
Not sure what causes this.

![image](https://github.com/alifeee/Factorio-Proximity-Voice-Chat/assets/13833017/faa85cc7-0808-4766-9243-36b1b4a45854)

**Comment by alifeee** [comment]
This is because the file is sometimes empty, so this code

https://github.com/alifeee/Factorio-Proximity-Voice-Chat/blob/777b5a9afe4cea9c597d7f10655a00d3459b535e/mumble/logfile.c#L258-L262

...returns an error, so Mumble thinks that the link has been lost.

I should handle "error 4" and give no information to Mumble, rather than saying it failed to grab info.

Perhaps I could just return `true`.

The documentation for `mumble_fetchPositionalData` suggests that this is the only valid option (if we can't fetch data but also don't want to stop tracking it)

https://github.com/alifeee/Factorio-Proximity-Voice-Chat/blob/777b5a9afe4cea9c597d7f10655a00d3459b535e/mumble/include/MumblePlugin_v_1_0_x.h#L208-L213

Alternatively, we could add a "retry" for when the positional data is blank (so it will wait until the next file is written).

**Comment by alifeee** [comment]
Returning true stops the link disconnecting (natually), but I do not know how Mumble will handle that we do not alter `avatarPos` before returning true.

These values might flick to 0 or garbage every now and then

![image](https://github.com/alifeee/Factorio-Proximity-Voice-Chat/assets/13833017/37311b06-5b6c-409a-8ae6-67a3a3c8df30)

**Comment by alifeee** [comment]
Okay, it seems fine. The values are left in memory from the previous time

![image](https://github.com/alifeee/Factorio-Proximity-Voice-Chat/assets/13833017/7b380959-2ce4-4c71-92da-db83faf86078)

```python
		char *log = malloc(150);
		sprintf(log, "FactorioPA error%d", error);
		sprintf(log, "%s %f %f %f", log, avatarPos[0], avatarPos[1], avatarPos[2]);
		mumbleAPI.log(ownID, log);
```

**Comment by alifeee** [comment]
(I think) this is fixed by release 0.6.0
